wpilibj/src/main/java/edu/wpi/first/wpilibj/RobotBase.java

```diff
@@ -382,8 +382,22 @@ private static <T extends RobotBase> void runRobot(Supplier<T> robotSupplier) {
       }
       String robotName = "Unknown";
       StackTraceElement[] elements = throwable.getStackTrace();
-      if (elements.length > 0) {
-        robotName = elements[0].getClassName();
+      for (var element : elements) {
+        try {
+          if (RobotBase.class.isAssignableFrom(Class.forName(element.getClassName()))) {
+            robotName = element.getClassName();
+            break;
+          }
+        } catch (ClassNotFoundException e) {
+          DriverStation.reportError(
+              "Could not find class from stack trace? This should not happen.\n"
+                  + "If simulating in VS Code, try opening the Command Palette "
+                  + "and running Clean Java Language Server Workspace.\n"
+                  + "If this is on a real robot, try redeploying. If this is persistent, "
+                  + "report this on https://github.com/wpilibsuite/allwpilib/issues/new?template=bug_report.md. Error: "
+                  + throwable,
+              elements);
+        }
```

Comment:
Do we really need to report an error for *every* class that couldn't be found in the stack trace?  Or only if robotName is blank at the end of the search?